### PR TITLE
Add support for Appium & ADB CLI

### PR DESCRIPTION
**Pull request by bgalamb**
Added support for Appium too.
Appium (and ADB) puts single extra params to the Intent . It is unable to add an entire bundle extra param.
Added support for StringExtra parameter which can be added successfully.

### CODE REVIEW

**Comment by iamolegga** [comment]
Hi, thanks for this. The one question I have, will data be consistent between two platforms? Asking because for ios there is parsing of passed string to map, but in this pr values passed as raw string, right?

**Comment by bgalamb** [comment]
Yes, you were right. I have refactored the code a little bit, so that all extensions of all types are put into the map as they are.
Except for the extension with key "launchArgs" which is handled as before (extracted keys and values)

File: README.md [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+Mostly it's made for using 
+* [`launchArgs parameter of device.launchApp method`](https://github.com/wix/Detox/blob/master/docs/APIRef.DeviceObjectAPI.md#7-additional-launch-arguments) of [Detox](https://github.com/wix/Detox/)
+* [`optionalIntentArguments (Android)` and `processArguments (iOS)`](http://appium.io/docs/en/writing-running-appium/caps/) parameters with [Appium](http://appium.io/)
 
 **iOS**: it takes data from `[[NSProcessInfo processInfo] arguments]`
 
-**Android**: it takes data from `currentActivity.getIntent().getBundleExtra("launchArgs")`
+**Android**: it takes data from [`currentActivity.getIntent().getBundleExtra("launchArgs")`](https://developer.android.com/reference/android/content/Intent#getBundleExtra(java.lang.String)) together with [`currentActivity.getIntent().getStringExtra("launchArgs")`](https://developer.android.com/reference/android/content/Intent#getStringExtra(java.lang.String))
+* see [`ADB docs`](https://developer.android.com/studio/command-line/adb#IntentSpec) for details
```
Comment (iamolegga) [inline README.md:12]: can you please leave the same code style (here and in .java file) as it was before, thanks

File: android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java [hunk 2]
```diff
@@ -30,20 +30,31 @@ public String getName() {
...
+                                Set < String > ks = bundle.keySet();
+                                Iterator < String > iterator = ks.iterator();
+                                while (iterator.hasNext()) {
+                                    String innerkey = iterator.next();
+                                    map.put(innerkey, bundle.getString(innerkey));
+                                }
+                            }
+                        } else {
+                            map.put(key, bundleExtras.get(key));
+                        }
```
Comment (iamolegga) [inline android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java:57]: I'm not sure if this is backward compatible with previous version, let's just leave previous version and add new logic separately?

File: android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java [hunk 3]
```diff
@@ -30,20 +30,31 @@ public String getName() {
     }
 
     @Override
-    public Map<String, Object> getConstants() {
-        Map<String, Object> map = new HashMap();
+    public Map < String, Object > getConstants() {
+        Map < String, Object > map = new HashMap();
```
Comment (iamolegga) [inline android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java:34]: is this kind of changes necessary?

**Review by iamolegga — CHANGES_REQUESTED** [review]
thanks, I have a few suggestion: let's make changes as small as possible without refactoring and by adding new logic in separate statement, also let's make it without unnecessary code style changes

File: android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java (hunk 3 above)
Comment (bgalamb) [reply android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java]: reverted back, code style error...

File: README.md (hunk 1 above)
Comment (bgalamb) [reply README.md]: ok. My codestyle was different. I reverted it .

File: android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java (hunk 2 above)
Comment (bgalamb) [reply android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java:56]: Yes, this is backward compatible, and needed too. This is the piece which out all key,values which are there in the intent  to the map. see https://developer.android.com/reference/android/content/Intent#getExtras() and https://stackoverflow.com/questions/6894513/how-to-get-all-extra-values-from-intent/35041128

**Comment by iamolegga** [comment]
Thanks again for this @bgalamb . Published in v2.0.0, please check it out 👍

**Pull request merged**